pages/development/theming.mdx

```diff
@@ -1,21 +1,270 @@
+// embeddable.theme.ts
+import { Theme, defineTheme } from 'vanilla-components';
+import lightTheme from './src/themes/light.theme';
+import darkTheme from './src/themes/dark.theme';
+
+interface ClientContext {
+  [key: string]: any;
+  theme: string;
+}
+
+const themeProvider = (clientContext: ClientContext, parentTheme: Theme): Theme => {
+  const { theme } = clientContext;
+  let currentTheme = lightTheme;
+  if (theme === 'dark') {
+    currentTheme = darkTheme;
+  }
+	return defineTheme(parentTheme, currentTheme);
+};
+
+export default themeProvider;
+```
+
+As you can see, we’re still merging themes with the default, but in this case we’re checking for a string value named `theme` passed in via Client Context. We then swap our current theme based on the string value and merge it into the default. On your application side, all you’d need to do is have a toggle that changes the Client Context of any Embeddables from `light` to `dark` to switch between the two.
+
+We encourage you to adjust the `ClientContext` interface to include any other values you’re passing in via Client Context, especially if you intend to use them in this file in any way (for example, you could pass a set of chart colors via Client Context and then plug them into the theme, for a hybrid between this approach and the one below).
+
+**Sending Theme Objects Via Client Context**
+
+If you’re going to support more than a handful of themes, for example, if you want to have a theme for each of your own clients, it will likely be a lot of work to create and support a large number of theme files. You may also already have theme data for clients stored in a database, or other location, and want to repurpose that data rather than manually crafting themes. In either case, the great thing about Client Context is it’s just arbitrary JSON, and our theme files are just a JavaScript object, so it’s very straightforward to combine the two. Here’s what it might look like:
+
+```tsx
+// your front-end code
+<em-beddable token={{securityToken}} clientContext={{
+  theme: {
+    brand: {
+      primary: '#0066FF',
+      secondary: '#FF6600',
+    },
+    font: {
+      family: 'Consolas',
+    },
+  }
+}} />
+```
+
+```tsx
+// embeddable.theme.ts
+import { Theme, defineTheme } from 'vanilla-components';
+
+const themeProvider = (clientContext: any, parentTheme: Theme): Theme => {
+  const { theme } = clientContext;
+	return defineTheme(parentTheme, theme);
+};
+
+export default themeProvider;
+```
+
+This approach would support as many themes as you’d like, and the theme objects you send can be a few overrides or all of the data needed to completely change every aspect of the theme! 
+
+### A Note About Custom Fonts
+
+Under the hood, our theme provider attempts to find any custom fonts you define in `theme.font.family` via Google Fonts. If they support it, we can load it. If you’re trying to use another font provider, please contact our Customer Success team for further information.
```

Comment:
@cwbuecheler could "Google Fonts" link to a relevant google fonts page, where the user can check whether their font is available?